crates/pathfinder/src/state/sync.rs

```diff
@@ -391,6 +369,45 @@ where
     }
 }
 
+async fn latest_n_blocks(
+    storage: Storage,
+    n: usize,
+) -> anyhow::Result<Vec<(BlockNumber, BlockHash, StateCommitment)>> {
+    tokio::task::spawn_blocking(move || {
+        let mut connection = storage
+            .connection()
+            .context("Creating database connection")?;
+        let tx = connection
+            .transaction()
+            .context("Creating database transaction")?;
+
+        let mut stmt = tx
+            .prepare_cached(
+                "SELECT number, hash, root FROM starknet_blocks ORDER BY number DESC LIMIT ?",
+            )
+            .context("Preparing database statement")?;
+        let rows = stmt
+            .query_map([n], |row| {
+                let number: BlockNumber = row.get(0).unwrap();
+                let hash: BlockHash = row.get(1).unwrap();
+                let commitment: StateCommitment = row.get(2).unwrap();
+
+                Ok((number, hash, commitment))
+            })
+            .context("Querying database")?;
+
+        let mut blocks = Vec::new();
+        for row in rows {
+            blocks.push(row.context("Reading row from database")?);
+        }
+        blocks.reverse();
```

Comment:
Why `ORDER BY DESC` and then reverse?

Reply:
I want the latest `N` values, but in ascending order. So `ORDER BY DESC` + `LIMIT N` gives me the latest `N` values, but in descending order. So I then reverse it again. Alternative is to first count, pre-allocate and then insert backwards but I didn't think the (low) frequency of this occuring is worth the extra loc.

Reply:
My bad, I didn't notice the limit.

Reply:
Does indicate its worth a comment pointing it out then.